go/vt/worker/split_clone_test.go

```diff
@@ -1027,6 +1029,10 @@ func TestSplitCloneV2_NoMasterAvailable(t *testing.T) {
 			}
 		}
 
+		// Also wait for explicit master demoted message.
+		// statsRetryCounters is not reliable.
```

Comment:
Which race gets solved here?

The stats counter should only be increased after the master failover was simulated. The only exception might be if the variable is already non-zero at the start? In that case, we should just look at the difference and not add another notification channel?

Reply:
This is a hard to reproduce race. The symptoms I saw were that the test would start writing to the leftReplica instead of leftMaster. Adding debug statements made the failure non-reproduceable. My theory was that the counter gets incremented prematurely, which causes leftReplica to be promoted immediately. After I added this channel, I couldn't reproduce the failure. So, I assumed the fix was right.

I'm open to alternate theories :).